compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/types/irTypePredicates.kt

```diff
@@ -29,55 +68,79 @@ private fun classFqNameEquals(symbol: IrClassSymbol, fqName: FqNameUnsafe): Bool
+            this[getPublicSignature(StandardNames.BUILT_INS_PACKAGE_FQ_NAME, primitiveType.typeName.asString())] = primitiveType
+        }
+    }
+
+private val primitiveArrayTypesSignatures: Map<PrimitiveType, IdSignature.PublicSignature> =
+    newHashMapWithExpectedSize<PrimitiveType, IdSignature.PublicSignature>(PrimitiveType.values().size).apply {
+        for (primitiveType in PrimitiveType.values()) {
+            this[primitiveType] =
+                getPublicSignature(StandardNames.BUILT_INS_PACKAGE_FQ_NAME, "${primitiveType.typeName.asString()}Array")
+        }
+    }
+
 private fun classFqNameEquals(declaration: IrClass, fqName: FqNameUnsafe): Boolean =
-    declaration.name == fqName.shortName() && fqName == declaration.fqNameWhenAvailable?.toUnsafe()
+    declaration.hasEqualFqName(fqName.toSafe())
 
-fun IrType.isAny(): Boolean = isNotNullClassType(StandardNames.FqNames.any)
-fun IrType.isNullableAny(): Boolean = isNullableClassType(StandardNames.FqNames.any)
+fun IrType.isAny(): Boolean = isNotNullClassType(IdSignatureValues.any)
+fun IrType.isNullableAny(): Boolean = isNullableClassType(IdSignatureValues.any)
 
-fun IrType.isString(): Boolean = isNotNullClassType(StandardNames.FqNames.string)
-fun IrType.isNullableString(): Boolean = isNullableClassType(StandardNames.FqNames.string)
-fun IrType.isStringClassType(): Boolean = isClassType(StandardNames.FqNames.string)
-fun IrType.isArray(): Boolean = isNotNullClassType(StandardNames.FqNames.array)
-fun IrType.isNullableArray(): Boolean = isNullableClassType(StandardNames.FqNames.array)
-fun IrType.isCollection(): Boolean = isNotNullClassType(StandardNames.FqNames.collection.toUnsafe())
-fun IrType.isNothing(): Boolean = isNotNullClassType(StandardNames.FqNames.nothing)
-fun IrType.isKClass(): Boolean = isNotNullClassType(StandardNames.FqNames.kClass)
+fun IrType.isString(): Boolean = isNotNullClassType(IdSignatureValues.string)
+fun IrType.isNullableString(): Boolean = isNullableClassType(IdSignatureValues.string)
+fun IrType.isStringClassType(): Boolean = isClassType(IdSignatureValues.string)
+fun IrType.isArray(): Boolean = isNotNullClassType(IdSignatureValues.array)
+fun IrType.isNullableArray(): Boolean = isNullableClassType(IdSignatureValues.array)
+fun IrType.isCollection(): Boolean = isNotNullClassType(IdSignatureValues.collection)
+fun IrType.isNothing(): Boolean = isNotNullClassType(IdSignatureValues.nothing)
+fun IrType.isKClass(): Boolean = isNotNullClassType(IdSignatureValues.kClass)
 
-fun IrType.isPrimitiveType(): Boolean = StandardNames.FqNames.fqNameToPrimitiveType.keys.any { isNotNullClassType(it) }
-fun IrType.isNullablePrimitiveType(): Boolean = StandardNames.FqNames.fqNameToPrimitiveType.keys.any { isNullableClassType(it) }
+fun IrType.isPrimitiveType(): Boolean = idSignatureToPrimitiveType.keys.any { isNotNullClassType(it) }
+fun IrType.isNullablePrimitiveType(): Boolean = idSignatureToPrimitiveType.keys.any { isNullableClassType(it) }
```

Comment:
This is optional and can be fixed separately, but maybe do something like this to avoid traversing the map?
```
classOrNull?.takeIf(IrClassSymbol::isPublicApi)?.signature in idSignatureToPrimitiveType
```
(but also handle nullability)